includes/tax-functions.php

```diff
@@ -442,3 +410,63 @@ function edd_download_is_tax_exclusive( $download_id = 0 ) {
 
 	return (Bool) apply_filters( 'edd_download_is_tax_exclusive', $ret, $download_id );
 }
+
+/**
+ * Gets the tax rate object from the database.
+ * Used in `edd_get_tax_rate`, `edd_build_order`, `edd_add_manual_order`.
+ *
+ * @param array $args
+ * @return object|boolean
+ *
+ * @since 3.0
+ */
+function edd_get_tax_rate_object( $args ) {
+
+	$rate = false;
+	if ( empty( $args['country'] ) ) {
+		return $rate;
+	}
+
+	// Fetch all the tax rates from the database.
+	// The region is not passed in deliberately in order to check for country-wide tax rates.
+	$tax_rates = edd_get_tax_rates(
+		array(
+			'name'   => $args['country'],
+			'status' => 'active',
+		),
+		OBJECT
+	);
+
+	// Save processing if only one tax rate is returned.
```

Comment:
Wondering if this section is really needed. Feels like unnecessary code duplication. If there's only one rate then the `foreach` loop would still handle it just fine.

Reply:
I wasn't sure about editing too much of the code in this as it was extracted from the original tax rate function, but I agree. Will remove.